Run gc when idle

eatkins opened the pull request:
I often find that when I run a command it takes a long time to start up
because sbt triggers a full gc. To improve the ux, I update the command
exchange to periodically run full gc while it's waiting for a command to
run.

Bonus: optimize imports

- [x] I've read the [CONTRIBUTING](https://github.com/sbt/sbt/blob/develop/CONTRIBUTING.md) guidelines

eatkins committed:
Run gc when idle

I often find that when I run a command it takes a long time to start up
because sbt triggers a full gc. To improve the ux, I update the command
exchange to run full gc only once while it's waiting for a command to
run and only after the user has been idle for at least one minute.

Bonus: optimize imports

eed3si9n commented on main/src/main/scala/sbt/Main.scala:
```diff
@@ -884,7 +885,8 @@ object BuiltinCommands {
     val s1 = exchange run s0
     exchange publishEventMessage ConsolePromptEvent(s0)
-    val exec: Exec = exchange.blockUntilNextExec
+    val minGCInterval = Project.extract(s1).getOpt(Keys.minForcegcInterval).getOrElse(1.minute)
```
We probably should use `GCUtil.defaultMinForcegcInterval` instead of `1.minute`.
We can also take this opportunity to bump the min interval.

eatkins replied:
Good points. I re-pushed with a value of 10.minutes, but I more or less plucked that out of thin air.

eatkins replied:
I also reworked it to defer the full GC until the user has been idle for at least one minute. It will also only force full GC at most once between tasks.

eed3si9n reviewed (changes requested):
minor suggestion.

eatkins committed:
Lint CommandExchange.scala

Clears out intellij warnings.

eatkins committed:
Reduce command polling period

The only thing this function does is poll a queue and check a deadline.
There is no need to put such a large sleep duration in.
